src/main/java/org/sunbird/profile/service/ProfileServiceImpl.java

```diff
@@ -338,6 +351,108 @@ public SBApiResponse profileUpdate(Map<String, Object> request, String userToken
 		return response;
 	}
 
+	private String validateCadreDetails(Map<String, Object> profileDetailsMap) {
+		List<Map<String, Object>> cadreDetails;
+
+		try {
+			if (profileDetailsMap.containsKey(Constants.CADRE_DETAILS)
+					&& profileDetailsMap.get(Constants.CADRE_DETAILS) != null) {
```

Comment:
we called this method bcz the given profileDetails had this cadreDetails in the requestBody... 
so simply get the object and check for null -- refer the comment for returning error messages...

Reply:
done